### ISSUE TITLE
Cannot import table from postgres

### ISSUE THREAD

**Issue by Kohlroulade** [koordinates/kart#444]
I´m trying to import some tables from a postgres-db. However when I call `kart import postgres://myuser:password@localhost/database` I get the following error:

```
Traceback (most recent call last):
  File "lib\site-packages\sqlalchemy\engine\base.py", line 3141, in _wrap_pool_connect
  File "lib\site-packages\sqlalchemy\pool\base.py", line 301, in connect
  File "lib\site-packages\sqlalchemy\pool\base.py", line 755, in _checkout
  File "lib\site-packages\sqlalchemy\pool\base.py", line 419, in checkout
  File "lib\site-packages\sqlalchemy\pool\impl.py", line 145, in _do_get
  File "lib\site-packages\sqlalchemy\util\langhelpers.py", line 72, in __exit__
  File "lib\site-packages\sqlalchemy\util\compat.py", line 211, in raise_
  File "lib\site-packages\sqlalchemy\pool\impl.py", line 142, in _do_get
  File "lib\site-packages\sqlalchemy\pool\base.py", line 247, in _create_connection
  File "lib\site-packages\sqlalchemy\pool\base.py", line 362, in __init__
  File "lib\site-packages\sqlalchemy\pool\base.py", line 605, in __connect
  File "lib\site-packages\sqlalchemy\util\langhelpers.py", line 72, in __exit__
  File "lib\site-packages\sqlalchemy\util\compat.py", line 211, in raise_
  File "lib\site-packages\sqlalchemy\pool\base.py", line 599, in __connect
  File "lib\site-packages\sqlalchemy\engine\create.py", line 578, in connect
  File "lib\site-packages\sqlalchemy\engine\default.py", line 548, in connect
  File "lib\site-packages\psycopg2\__init__.py", line 127, in connect
psycopg2.OperationalError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "kart_cli.py", line 4, in <module>
  File "kart\cli.py", line 334, in entrypoint
  File "lib\site-packages\click\core.py", line 829, in __call__
  File "lib\site-packages\click\core.py", line 782, in main
  File "kart\cli.py", line 158, in invoke
  File "lib\site-packages\click\core.py", line 1259, in invoke
  File "lib\site-packages\click\core.py", line 1066, in invoke
  File "lib\site-packages\click\core.py", line 610, in invoke
  File "lib\site-packages\click\decorators.py", line 21, in new_func
  File "kart\init.py", line 449, in init
  File "kart\sqlalchemy_import_source.py", line 124, in get_tables
  File "lib\site-packages\sqlalchemy\engine\base.py", line 3096, in connect
  File "lib\site-packages\sqlalchemy\engine\base.py", line 92, in __init__
  File "lib\site-packages\sqlalchemy\engine\base.py", line 3175, in raw_connection
  File "lib\site-packages\sqlalchemy\engine\base.py", line 3146, in _wrap_pool_connect
  File "lib\site-packages\sqlalchemy\engine\base.py", line 2005, in _handle_dbapi_exception_noconnection
  File "lib\site-packages\sqlalchemy\util\compat.py", line 211, in raise_
  File "lib\site-packages\sqlalchemy\engine\base.py", line 3142, in _wrap_pool_connect
  File "lib\site-packages\sqlalchemy\pool\base.py", line 301, in connect
  File "lib\site-packages\sqlalchemy\pool\base.py", line 761, in _checkout
  File "lib\site-packages\sqlalchemy\pool\base.py", line 419, in checkout
  File "lib\site-packages\sqlalchemy\pool\impl.py", line 145, in _do_get
  File "lib\site-packages\sqlalchemy\util\langhelpers.py", line 72, in __exit__
  File "lib\site-packages\sqlalchemy\util\compat.py", line 211, in raise_
  File "lib\site-packages\sqlalchemy\pool\impl.py", line 142, in _do_get
  File "lib\site-packages\sqlalchemy\pool\base.py", line 247, in _create_connection
  File "lib\site-packages\sqlalchemy\pool\base.py", line 362, in __init__
  File "lib\site-packages\sqlalchemy\pool\base.py", line 605, in __connect
  File "lib\site-packages\sqlalchemy\util\langhelpers.py", line 72, in __exit__
  File "lib\site-packages\sqlalchemy\util\compat.py", line 211, in raise_
  File "lib\site-packages\sqlalchemy\pool\base.py", line 599, in __connect
  File "lib\site-packages\sqlalchemy\engine\create.py", line 578, in connect
  File "lib\site-packages\sqlalchemy\engine\default.py", line 583, in connect
  File "lib\site-packages\psycopg2\__init__.py", line 127, in connect
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError)
(Background on this error at: http://sqlalche.me/e/14/e3q8)
[3044] Failed to execute script kart_cli

```
I´m using the latest release of kart (0.9.0). My database is Postgres v13, my JAVA-Version is JRE 1.8

**Comment by rcoup** [comment]
Hi @Kohlroulade,

Is it possible to try a recent master build? (eg: download an installer from https://github.com/koordinates/kart/actions/runs/937934592) There have been a few database-related changes recently, and it'd help narrow it down.

While the error doesn't display anything super useful,`psycopg2.OperationalError` in `connect()` suggests your database might not be accessible. Can you double-check `psql -h localhost -U myuser database` works ok?

**Comment by Kohlroulade** [comment]
Updated to the mentioned kart-version without any change. 
Connection using psql worked as intended:
![image](https://user-images.githubusercontent.com/15833126/122031517-1342c780-cdcf-11eb-88c3-e4ed194edadd.png)

**Comment by craigds** [comment]
Hi @Kohlroulade 

1. with `postgres://` scheme I get a different error; I presume your error was actually with `postgresql://`? 

```
$ kart import postgres://myuser:password@localhost/database
Error: 'postgres://myuser:password@localhost/database' doesn't appear to be valid (tried formats: GPKG,PG,SHP)
postgres://myuser:password@localhost/database: No such file or directory
```

2. What operating system are you using; is that Windows 10?
3. Does adding `-vv` (`kart -vv import ...`) add any more context to this error?


I find it curious that there is no contextual message for the OperationalError at all (maybe this is just always how it behaves on windows; I haven't tested that yet). On my macOS I get a similar traceback but with a somewhat more meaningful error message (there's no database running, hence the error for me):


```
$ kart import postgresql://myuser:password@localhost/database
Traceback (most recent call last):
  File "sqlalchemy/engine/base.py", line 3142, in _wrap_pool_connect
  File "sqlalchemy/pool/base.py", line 301, in connect
  File "sqlalchemy/pool/base.py", line 761, in _checkout
  File "sqlalchemy/pool/base.py", line 419, in checkout
  File "sqlalchemy/pool/impl.py", line 145, in _do_get
  File "sqlalchemy/util/langhelpers.py", line 72, in __exit__
  File "sqlalchemy/util/compat.py", line 211, in raise_
  File "sqlalchemy/pool/impl.py", line 142, in _do_get
  File "sqlalchemy/pool/base.py", line 247, in _create_connection
  File "sqlalchemy/pool/base.py", line 362, in __init__
  File "sqlalchemy/pool/base.py", line 605, in __connect
  File "sqlalchemy/util/langhelpers.py", line 72, in __exit__
  File "sqlalchemy/util/compat.py", line 211, in raise_
  File "sqlalchemy/pool/base.py", line 599, in __connect
  File "sqlalchemy/engine/create.py", line 578, in connect
  File "sqlalchemy/engine/default.py", line 583, in connect
  File "psycopg2/__init__.py", line 127, in connect
psycopg2.OperationalError: could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432?


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "kart_cli.py", line 4, in <module>
  File "kart/cli.py", line 334, in entrypoint
  File "click/core.py", line 829, in __call__
  File "click/core.py", line 782, in main
  File "kart/cli.py", line 158, in invoke
  File "click/core.py", line 1259, in invoke
  File "click/core.py", line 1066, in invoke
  File "click/core.py", line 610, in invoke
  File "click/decorators.py", line 21, in new_func
  File "kart/init.py", line 265, in import_
  File "kart/import_source.py", line 221, in prompt_for_table
  File "kart/sqlalchemy_import_source.py", line 124, in get_tables
  File "sqlalchemy/engine/base.py", line 3096, in connect
  File "sqlalchemy/engine/base.py", line 92, in __init__
  File "sqlalchemy/engine/base.py", line 3175, in raw_connection
  File "sqlalchemy/engine/base.py", line 3146, in _wrap_pool_connect
  File "sqlalchemy/engine/base.py", line 2005, in _handle_dbapi_exception_noconnection
  File "sqlalchemy/util/compat.py", line 211, in raise_
  File "sqlalchemy/engine/base.py", line 3142, in _wrap_pool_connect
  File "sqlalchemy/pool/base.py", line 301, in connect
  File "sqlalchemy/pool/base.py", line 761, in _checkout
  File "sqlalchemy/pool/base.py", line 419, in checkout
  File "sqlalchemy/pool/impl.py", line 145, in _do_get
  File "sqlalchemy/util/langhelpers.py", line 72, in __exit__
  File "sqlalchemy/util/compat.py", line 211, in raise_
  File "sqlalchemy/pool/impl.py", line 142, in _do_get
  File "sqlalchemy/pool/base.py", line 247, in _create_connection
  File "sqlalchemy/pool/base.py", line 362, in __init__
  File "sqlalchemy/pool/base.py", line 605, in __connect
  File "sqlalchemy/util/langhelpers.py", line 72, in __exit__
  File "sqlalchemy/util/compat.py", line 211, in raise_
  File "sqlalchemy/pool/base.py", line 599, in __connect
  File "sqlalchemy/engine/create.py", line 578, in connect
  File "sqlalchemy/engine/default.py", line 583, in connect
  File "psycopg2/__init__.py", line 127, in connect
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432?

(Background on this error at: http://sqlalche.me/e/14/e3q8)
[8065] Failed to execute script kart_cli
```

**Comment by Kohlroulade** [comment]
Oh sorry, yes you´re right, the protocoll surely is postgresql`, not just `postgres`. 

So I tried this with the exact same error-message:

```
kart -vv import postgresql://myuser:password@localhost/database
```

**Comment by olsen232** [comment]
Hi @Kohlroulade,
sorry for the delay. Sorry, it seems the Postgres working copy is broken on 0.9.0 on Windows - the good news is that it is fixed in the now-latest version of Kart - 0.10.x. Let me know if that version works for you.
Thanks for trying out Kart and helping us improve it! Our cross platform regression testing could use some improvement.
